### PR TITLE
SYS-2748 adding happy path test for payout logic

### PR DESCRIPTION
**Pull request by nahuseyoum**
This PR build on top of [PR#12](https://github.com/Aventus-Network-Services/avn-parachain/pull/12) and adds a happy path test for our payout logic. This is an end-to-end test that covers the functionality of collators being rewarded instead of being a "traditional" unit test. Subsequent PRs will add unit tests for "bad" cases.

### CODE REVIEW

**Review by thadouk — DISMISSED** [review]

File: pallets/parachain-staking/src/test_reward_payout.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+use sp_runtime::traits::SignedExtension;
+
+#[test]
+fn end_to_end_happy_path() {
+    let collator1 = 1;
+    let collator2 = 2;
+    let nominator4 = 4;
+    let tx_sender = 3;
+    let era_blocks_have_been_authored = 1;
+    let expected_tx_fee: u128 = (BASE_FEE + TX_LEN as u64) as u128;
```
[the remark this answers is not in the document]
Comment (nahuseyoum) [reply pallets/parachain-staking/src/test_reward_payout.rs:20]: These are `test` values.

File: pallets/parachain-staking/src/lib.rs [hunk 2]
```diff
@@ -1349,6 +1351,12 @@ pub mod pallet {
 						account: to.clone(),
 						rewards: amt,
 					});
+
+					// Update storage with the amount we paid
+					<LockedEraPayout<T>>::mutate(|p| {
```
[the remark this answers is not in the document]
Comment (nahuseyoum) [reply pallets/parachain-staking/src/lib.rs:1356]: All the existing code uses tabs and I will run the formater later to update them all. New code files we push uses spaces.

**Review by alexpArtos — DISMISSED** [review]
I am approving after detailed walk-through.

**Review by alexpArtos — APPROVED** [review]

File: pallets/parachain-staking/src/test_reward_payout.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (thadouk) [reply pallets/parachain-staking/src/test_reward_payout.rs:16]: @alexpArtos the comment was added on a review that was approving the PR and was aligned with the discussion we had about code reviews.
The project is setup in a way, that dismisses approval reviews when new changes are pushed.

**Review by thadouk — APPROVED** [review]

**Pull request merged**